internal/lifecycle/lifecycle_test.go

```diff
@@ -220,6 +225,40 @@ func TestLifecycleStop(t *testing.T) {
 	})
 }
 
+func TestHookRecordsFormat(t *testing.T) {
+	t.Run("SortRecords", func(t *testing.T) {
+		r := make(HookRecords, 2)
+		t1, _ := time.ParseDuration("10ms")
+		t2, _ := time.ParseDuration("20ms")
+		f := fxreflect.Frame{
+			Function: "someFunc",
+			File:     "somefunc.go",
+			Line:     1,
+		}
+		r[0] = HookRecord{
+			CallerFrame: f,
+			Func: func(context.Context) error {
+				return nil
+			},
+			Runtime: t1,
+		}
+		r[1] = HookRecord{
+			CallerFrame: f,
+			Func: func(context.Context) error {
+				return nil
+			},
+			Runtime: t2,
+		}
+		sort.Sort(r)
+		for _, s := range []string{fmt.Sprintf("%v", r), fmt.Sprintf("%+v", r), fmt.Sprintf("%s", r)} {
```

Comment:
optional: seeing as the thing that's different in each case is the format, this could be `[]string{"%v", "%+v", and "%s"}` and the loop body can `fmt.Sprintf` on it.